packages/components/src/range-control/stories/index.tsx

```diff
@@ -0,0 +1,248 @@
+/**
+ * External dependencies
+ */
+import type { ComponentMeta, ComponentStory } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import { starEmpty, starFilled, styles, wordpress } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import RangeControl from '..';
+
+const ICONS = { starEmpty, starFilled, styles, wordpress };
+
+const meta: ComponentMeta< typeof RangeControl > = {
+	component: RangeControl,
+	title: 'Components/RangeControl',
+	argTypes: {
+		afterIcon: {
+			control: { type: 'select' },
+			options: Object.keys( ICONS ),
+			mapping: ICONS,
+		},
+		allowReset: { control: { type: 'boolean' } },
+		beforeIcon: {
+			control: { type: 'select' },
+			options: Object.keys( ICONS ),
+			mapping: ICONS,
+		},
+		color: { control: { type: 'color' } },
+		disabled: { control: { type: 'boolean' } },
+		help: { control: { type: 'text' } },
+		initialPosition: { control: { type: 'number' } },
+		marks: { control: { type: 'object' } },
+		min: { control: { type: 'number' } },
+		max: { control: { type: 'number' } },
+		onChange: { action: 'onChange', control: { type: null } },
+		railColor: { control: { type: 'color' } },
+		showTooltip: { control: { type: 'boolean' } },
+		step: { control: { type: 'number' } },
+		trackColor: { control: { type: 'color' } },
+		withInputField: { control: { type: 'boolean' } },
```

Comment:
A bunch of these can be removed, now that we have type data:

```suggestion
		beforeIcon: {
			control: { type: 'select' },
			options: Object.keys( ICONS ),
			mapping: ICONS,
		},
		color: { control: { type: 'color' } },
		help: { control: { type: 'text' } },
		marks: { control: { type: 'object' } },
		onChange: { action: 'onChange' },
		railColor: { control: { type: 'color' } },
		step: { control: { type: 'number' } },
		trackColor: { control: { type: 'color' } },
```

On the other hand, I guess we should disable the controls for these new callback props: `onFocus`, `onMouseLeave`, `onMouseMove`, `renderTooltipContent`. (Storybook disables controls on `on*` callback props by default, but in these cases I think it doesn't understand the types as function types?)

Also, we should disable the `value` control now that our stories are controlled.

Reply:
Thanks, I've cleaned up the controls here. ( https://github.com/WordPress/gutenberg/pull/41444/commits/9631a22b8d87b28f5285cf8153af2c687147d608 & https://github.com/WordPress/gutenberg/pull/41444/commits/7e06827ab81a020f81106ced4fd8283048a32610)